menuinst/utils.py

```diff
@@ -246,6 +245,52 @@ def deep_update(mapping: Mapping, *updating_mappings: Iterable[Mapping]) -> Mapp
     return updated_mapping
 
 
+def needs_admin(target_prefix: os.PathLike, base_prefix: os.PathLike) -> bool:
+    """
+    Checks if the current installation needs admin permissions.
+    """
+    if user_is_admin():
+        return False
+
+    if Path(target_prefix, ".nonadmin").exists():
+        # This file is planted by the constructor installer
+        # and signals we don't need admin permissions
+        return False
+
+    try:
+        Path(target_prefix, ".nonadmin").touch()
+        return False
```

Comment:
The assumption here is that if I can write into the directory, this is a non-admin installation to begin with, that file just hasn't been written before for some reason?

The `user_is_admin` check should protect from any kind of problematic permanent changes, but I'm wondering if that file should be cleaned up afterwards.

Reply:
Yep, we should only expect a `.nonadmin` file in the base environment of Windows installations. It won't exist anywhere else. What we are doing here is to try to create that file in the target environment (because that's where the menuinst JSON came from) for the first time. If we succeed, then we must be running in user mode, and we leave it there so we don't have to check in future runs. We do assume that permissions won't change in the future, and that other users will not try to write to it if they don't have permissions (I'm thinking HPC setups with an admin-protected base, but user-writable envs in their home folders).

Reply:
> We do assume [...] that other users will not try to write to it if they don't have permissions 

That's why I'm wondering if we should remove touched files to be safe

Reply:
Maybe only if they are old enough? Let's say at least a day old? Otherwise we will be creating and recreating files everytime `menuinst` is run, which can be several times per `conda` command.

Reply:
After thinking about this more, I think the risk is minimal. If you can write to the directory during the installation, you can do much worse than create shortcuts. And the presence of a `.noadmin` file really only tells `menuinst` to attempt to do a non-admin installation